languages/rust/exercises/concept/structs/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+#[derive(Clone, Debug)]
+pub struct User {
+    name: String,
+    age: u32,
+    weight: f32,
+}
+
+impl User {
+    pub fn new(name: String, age: u32, weight: f32) -> Self {
+        unimplemented!()
+    }
+
+    pub fn get_name(&self) -> &str {
+        unimplemented!()
+    }
+
+    pub fn get_age(&self) -> u32 {
+        unimplemented!()
+    }
+
+    pub fn get_weight(&self) -> f32 {
+        unimplemented!()
+    }
+
+    pub fn update_name(&mut self, new_name: String) {
+        unimplemented!()
+    }
+
+    pub fn update_age(&mut self, new_age: u32) {
```

Comment:
The idiomaticity argument is weaker in this case, but IMO `set_x` is more conventional than `update_x`.

```suggestion
    pub fn set_age(&mut self, new_age: u32) {
```